### PR TITLE
Add higher order list essentials

### PR DESCRIPTION
**Pull request by oleduc**

### CODE REVIEW

File: specs/array/filter.spec.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import assert from 'assert'
+import Id from '../../package/id'
+import filter from '../../package/filter'
```
Comment (pre63) [inline specs/array/filter.spec.js]: `import filter from '../../package/array/filter'`

File: webpack.config.js [hunk 2]
```diff
@@ -13,7 +13,9 @@ module.exports = {
     frozenArray: './frozenArray/index.js',
     future: './future/index.js',
     id: './id/index.js',
-    map: './map/index.js',
+    map: './array/map/index.js',
```
Comment (pre63) [inline webpack.config.js]: 👍

File: src/array/filter/index.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import safe from 'safe'
+
+// filter :: ƒ -> ([Any] -> [Any])
+export default func => safe(l => l.filter(func))
```
Comment (pre63) [inline src/array/filter/index.js]:
```
// reduce :: (ƒ, Any) -> ([Any] -> Any)
export default (func, acc) => safe(l => l.reduce(func, acc))
```

File: src/array/reduce/index.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import safe from 'safe'
+
+// reduce :: ƒ -> ([Any] -> Any -> Any)
+export default acc => func => safe(l => l.reduce(func, acc))
```
Comment (pre63) [inline src/array/reduce/index.js]:
```
// reduce :: (ƒ, Any) -> ([Any] -> Any)
export default (func, acc) => safe(l => l.reduce(func, acc))
```

File: specs/array/reduce.spec.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import assert from 'assert'
+import Id from '../../package/id'
+import reduce from '../../package/reduce'
```
Comment (pre63) [inline specs/array/reduce.spec.js]: import filter from '../../package/array/reduce'

**Comment by jfdoube** [comment]
First official PR. We will drink to that 🎉

File: specs/filter.spec.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import assert from 'assert'
+import Id from '../package/id'
+import filter from '../package/filter'
+
+describe('A filter', () =>
+  it('shoud filter arrays', () =>
+    Id([0, 1, 2, 3])
+      .map(filter(x => x >= 2))
```
Comment (jfdoube) [inline specs/filter.spec.js]: Can't that be chain ?

**Comment by ysael** [comment]
wicked @oleduc

**Pull request merged**